pkg/cmd/cli/describe/projectstatus.go

```diff
@@ -581,3 +569,94 @@ func describeServicePorts(spec kapi.ServiceSpec) string {
 		return " ports " + strings.Join(pairs, ", ")
 	}
 }
+
+func loadServices(g osgraph.Graph, graphLock sync.Mutex, namespace string, kclient kclient.Interface, client client.Interface) error {
+	svcs, err := kclient.Services(namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	graphLock.Lock()
+	defer graphLock.Unlock()
```

Comment:
They're not expensive relative to the network cal we're about to make. Deferring immediately after locking gives good locality of reference, ensures the unlock is called even in case of a panic, etc